src/components/cc-pricing-product/cc-pricing-product.js

```diff
@@ -67,163 +69,455 @@ export class CcPricingProduct extends LitElement {
+    const productName = this.product.name;
+    dispatchCustomEvent(this, 'add-plan', { productName, ...plan });
+  }
 
-          ${skeleton && !this.error ? html`
-            <slot>
-              <div>
-                <span class="description skeleton">${description}</span>
-              </div>
-            </slot>
-          ` : ''}
-          ${!skeleton && !this.error ? html`
-            <slot>${description}</slot>
-          ` : ''}
+  willUpdate (changedProperties) {
 
-        </div>
-      </slot>
+    // Everytime the product state changes, update the sorted plan list
+    if (changedProperties.has('product') && Array.isArray(this.product.planList)) {
+      this._sortedPlanList = this.product.planList
+        .sort((a, b) => a.price - b.price);
+    }
+
+    // Everytime the product state changes, update the filtered feature list
+    if (changedProperties.has('product') && Array.isArray(this.product.featureList)) {
+      this._featureList = this.product.featureList
+        .filter((feature) => AVAILABLE_FEATURES.includes(feature.code));
+    }
+  }
+
+  render () {
+
+    return html`
 
-      ${this.error ? html`
-        <cc-error>${i18n('cc-pricing-product.error')}</cc-error>
+      ${this.product.state === 'error' ? html`
+        <cc-notice intent="danger" message=${i18n('cc-pricing-product.error')}></cc-notice>
       ` : ''}
-      ${skeleton && !this.error ? html`
+      ${this.product.state === 'loading' ? html`
         <cc-loader></cc-loader>
       ` : ''}
-      ${!skeleton && !this.error ? html`
-        <cc-pricing-table
-          class="pricing-table"
-          .plans=${this.plans}
-          .features=${this.features}
-          .currency=${this.currency}
-          .temporality=${this.temporality}
-          action=${this.action}
-          @cc-pricing-table:add-plan=${this._onAddPlan}
-        ></cc-pricing-table>
-      ` : ''}
+      ${this.product.state === 'loaded' ? this._renderProductPlans() : ''}
+    `;
+  }
+
+  _renderProductPlans () {
+    // We don't really have a good way to detect when the component should switch between bit and small mode.
+    // Also, when this component is used several times in the page, it's better if all instances switch at the same breakpoint.
+    // 800 seems like a good arbitrary value for the content we need to display.
+    return (this._size > 800)
+      ? this._renderBigPlans()
+      : this._renderSmallPlans();
+  }
+
+  _renderBigPlans () {
+    const temporality = this.temporalityList ?? DEFAULT_TEMPORALITY_LIST;
+    const productName = this.product.name;
+
+    return html`
+      <table>
+        <caption class="visually-hidden">${this.product.name}</caption>
+        <thead>
+          <tr>
+            <th>${i18n('cc-pricing-product.plan')}</th>
+            ${this._featureList.map((feature) => html`
+              <th class=${classMap({ 'number-align': NUMBER_FEATURE_TYPES.includes(feature.type) })}>${this._getFeatureName(feature)}</th>
```

Comment:
Not really sold on this idea but either way is fine with me.

Reply:
I feel like it will end up the same (unless I misunderstood):
```js
_isFeatureNumber (featureType) {
  return NUMBER_FEATURE_TYPES.includes(feature.type);
}
...
<th class=${classMap({ 'number-align': this._isFeatureNumber(feature.type) })}>${this._getFeatureName(feature)}</th>

```